### PR TITLE
Unblock Storybook-in-tests with l10n

### PR DESCRIPTION
**Pull request by Vinnl**
Accessing next/headers in tests results in an error because it's not in a Server Component context, so calling our regular `getL10n()` functions from inside Storybook would result in test failures. Instead, I replaced that call with a function that just loads the English strings directly, rather than looking at request headers.

Note that there are also a couple of test changes - now that proper English strings are loaded, some of the tests would fail without accommodating that :)

I've set https://github.com/mozilla/blurts-server/pull/3209's branch as the target, because this also resolves what I mentioned there:

> Note that unfortunately, I wasn't able to re-use the Storybook story in the unit test

### CODE REVIEW

File: locales-pending/premium.ftl [hunk 1]
```diff
@@ -13,6 +13,8 @@ main-nav-link-dashboard-label = Dashboard
 main-nav-link-faq-label = FAQs
 main-nav-link-faq-tooltip = Frequently asked questions
 
+mobile-menu-label = Main menu
```
Comment (Vinnl) [inline locales-pending/premium.ftl:16]: I added this because the a11y test now caught that the mobile menu did not have a unique label.

File: src/app/(proper_react)/(onboarding)/user/welcome/Onboarding.stories.tsx [hunk 2]
```diff
@@ -14,5 +14,5 @@ export default meta;
 type Story = StoryObj<typeof OnboardingEl>;
 
 export const Onboarding: Story = {
-  render: () => <OnboardingEl user={{}} />,
+  render: () => <OnboardingEl user={{ email: "example@example.com" }} />,
```
Comment (Vinnl) [inline src/app/(proper_react)/(onboarding)/user/welcome/Onboarding.stories.tsx:17]: Sorry, this is an unrelated change I snuck in - I saw a red squiggly that this mock data did not include an email address.

File: src/app/functions/server/mockL10n.ts
Comment (Vinnl) [inline src/app/functions/server/mockL10n.ts:1]: This is a bit of a complex file, but basically it's a copy of the `l10n.ts` file in the same dir, but modified to always load the `en` locales without checking request headers (which are not available in tests), and to use `require.context` for Storybook (which uses Webpack, which provides that function) and `node:fs` in Jest (which runs in a Node context).

File: .storybook/preview.tsx [hunk 3]
```diff
@@ -44,9 +30,7 @@ const AppDecorator: Exclude<Preview["decorators"], undefined>[0] = (
 
   return (
     <L10nProvider bundleSources={l10nBundles}>
-      <ReactAriaI18nProvider locale={getLocale(l10nBundles)}>
-        {storyFn()}
-      </ReactAriaI18nProvider>
+      <ReactAriaI18nProvider locale={"en"}>{storyFn()}</ReactAriaI18nProvider>
```
Comment (pdehaan) [inline .storybook/preview.tsx:33]: Not sure it needs the curly braces in `locale={"en"}`:
```suggestion
      <ReactAriaI18nProvider locale="en">{storyFn()}</ReactAriaI18nProvider>
```

File: src/app/(proper_react)/(onboarding)/user/welcome/Onboarding.test.tsx [hunk 4]
```diff
@@ -33,7 +33,7 @@ it("can open a dialog with more information", async () => {
   expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
 
   const explainerTrigger = screen.getByRole("button", {
-    name: "onboarding-get-started-content-explainer",
+    name: "See how we protect your data",
```
[the remark this answers is not in the document]
Comment (Vinnl) [reply src/app/(proper_react)/(onboarding)/user/welcome/Onboarding.test.tsx:36]: Yeah, it would. (That was already the case, though, as string changes would also require a new l10n ID.) [The idea](https://testing-library.com/docs/queries/about#priority) is that we simulate what the user would do, so that e.g. if a "Sign up" button were to suddenly say "Sign in", the tests break because the user's expected flow would break.

It's not that much work to fix since the test runner should clearly indicate which tests fail, and strings don't change that often, so IMHO that's a reasonable trade-off. (But: definitely something we could discuss in the tech meeting if you're interested.)

File: .storybook/preview.tsx (hunk 3 above)
Comment (Vinnl) [reply .storybook/preview.tsx]: Fairly sure it does not: 95f9e8acf.
